jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/preprocessor/UserAndGroupsTasksPreprocessor.java

```diff
@@ -77,8 +78,11 @@ public void preprocess(DataSetLookup lookup) {
                         addUserAndGroupsFromIdentityProvider(orgEntities, potOwner);
                     }
 
-                    it.remove();
                 }
+
+                // we need to look for recursively for the filter
+                removeCoreFunctionFilter(lookup.getFirstFilterOp().getColumnFilterList(), column);
```

Comment:
shouldn't this be placed in exact same location as 'it.remove()' before?

Reply:
you are right.
changed it.